pkg/test/framework/components/echo/instance.go

```diff
@@ -29,14 +29,25 @@ import (
 type Protocol string
 
 const (
-	// EchoProtocolHTTP calls echo with HTTP
-	EchoProtocolHTTP Protocol = "http"
-	// EchoProtocolGRPC calls echo with GRPC
-	EchoProtocolGRPC Protocol = "grpc"
-	// EchoProtocolWebSocket calls echo with WebSocket
-	EchoProtocolWebSocket Protocol = "ws"
+	// HTTP calls echo with HTTP
+	HTTP Protocol = "http"
+
+	// GRPC calls echo with GRPC
+	GRPC Protocol = "grpc"
+
+	// WebSocket calls echo with WebSocket
+	WebSocket Protocol = "ws"
 )
 
+func (p Protocol) normalize() Protocol {
+	switch p {
+	case HTTP, GRPC, WebSocket:
+		return p
+	default:
```

Comment:
how about https or tcp?
It is not clear what normalize is normalizing. It looks like this is for URLs only?

Reply:
These are what the application currently supports.